store/models/user.go

```diff
@@ -15,10 +16,21 @@ type User struct {
 	CreatedAt      Time   `json:"createdAt" storm:"index"`
 }
 
+// https://davidcel.is/posts/stop-validating-email-addresses-with-regex/
+var emailRegexp = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
 // NewUser creates a new user by hashing the passed plainPwd with bcrypt.
 func NewUser(email, plainPwd string) (User, error) {
-	if len(email) == 0 || len(plainPwd) == 0 {
-		return User{}, errors.New("Must enter an email or password")
+	if len(email) == 0 {
+		return User{}, errors.New("Must enter an email")
+	}
+
+	if !emailRegexp.MatchString(email) {
+		return User{}, errors.New("Invalid email format")
+	}
+
+	if len(plainPwd) < 8 || len(plainPwd) > 1028 {
```

Comment:
How do you feel about making these values config variables?

Reply:
Let's add this as a follow up story.

Reply:
https://www.pivotaltracker.com/story/show/159445522